File: backend/src/domain.rs
use serde::Serialize;

/// Espelha `EventMetadata` do contrato.
#[derive(Clone, Debug, Serialize, PartialEq, Eq)]
pub struct EventMetadata {
    pub name: String,
    pub description: String,
    pub image_ipfs: String,
}

/// Evento + id hex (32 bytes).
#[derive(Clone, Debug, Serialize, PartialEq, Eq)]
pub struct Event {
    /// `event_id` em hex (64 chars).
    pub id: String,
    #[serde(flatten)]
    pub metadata: EventMetadata,
}

File: backend-go/domain.go
package main

// EventMetadata mirrors the contract's EventMetadata.
type EventMetadata struct {
	Name        string `json:"name"`
	Description string `json:"description"`
	ImageIpfs   string `json:"image_ipfs"`
}

// Event is an event id (hex) plus its metadata (flattened in JSON).
type Event struct {
	Id string `json:"id"`
	EventMetadata
}
